pkg/client/pods.go

```diff
@@ -73,6 +74,11 @@ func (c *pods) Delete(name string) error {
 	return c.r.Delete().Namespace(c.ns).Resource("pods").Name(name).Do().Error()
 }
 
+// Same as above but with pod termination grace period specified.
+func (c *pods) DeleteGracefully(name string, gracePeriod uint64) error {
```

Comment:
Instead of adding a new method, refactor Delete to take api.DeleteOptions.  In this method, call `obj, err := api.Scheme.ConvertToVersion(deleteOptions, c.RESTClient.APIVersion)`, then call api.Convert(obj, map[string]string)`